openwebstart/src/main/java/com/openwebstart/jvm/LocalRuntimeManager.java

```diff
@@ -95,7 +96,15 @@ private void saveRuntimes() throws IOException {
             final File jsonFile = new File(cachePath, RuntimeManagerConstants.JSON_STORE_FILENAME);
             if (jsonFile.exists()) {
                 if (!jsonFile.delete()) {
-                    throw new IOException("Unable to delete old config file!");
+                    // if the file is locked, try again after sometime
+                    LOG.debug("Could not delete {}. File maybe locked. Trying again.", RuntimeManagerConstants.JSON_STORE_FILENAME);
+                    try {
+                        TimeUnit.SECONDS.sleep(1);
+                    } catch (InterruptedException e) {
+                    }
+                    if (!jsonFile.delete()) {
```

Comment:
can you check for `jsonFile.exists() && !jsonFile.delete()`

Otherwise it looks good. Why is it still in draft mode?